crates/ty_python_semantic/src/types.rs

```diff
@@ -8319,10 +8354,21 @@ impl<'db> ModuleLiteralType<'db> {
             }
         }
 
-        self.module(db)
+        let result = self
+            .module(db)
             .file(db)
             .map(|file| imported_symbol(db, file, name, None))
-            .unwrap_or_default()
+            .unwrap_or_default();
+
+        // If the normal lookup failed, try to call the module's __getattr__ function
+        if result.place.is_unbound() {
+            let getattr_result = self.try_module_getattr(db, name);
+            if !getattr_result.place.is_unbound() {
+                return getattr_result;
+            }
```

Comment:
There's really no point in checking for unbound here, since we are only here in the first place if `result` is unbound, and one unbound is as good as another.
```suggestion
            return self.try_module_getattr(db, name);
```